packages/android-app/android/xnative/build.gradle

```diff
@@ -0,0 +1,96 @@
+apply plugin: 'com.android.library'
+apply plugin: 'maven-publish'
+import groovy.json.JsonSlurper
+
+def xnativeVersion() {
+    def contents = rootProject.file('../package.json').text
+    def jsonSlurper = new JsonSlurper()
+    jsonSlurper.parseText(contents).version
+}
+
+def reactRepositoryPath = "$rootDir/../assets/react-native-modules"
+
+task sourceJar(type: Jar) {
+    from android.sourceSets.main.java.srcDirs
+    classifier "source"
+}
+
+publishing {
+    publications {
+        xnative(MavenPublication) {
+            groupId 'uk.co.thetimes'
+            artifactId 'xnative'
+            version xnativeVersion()
+            artifact(sourceJar)
+            artifact("$buildDir/outputs/aar/xnative-release.aar")
+            pom.withXml {
+                final dependenciesNode = asNode().appendNode('dependencies')
+
+                ext.addDependency = { Dependency dep, String scope ->
+                    if (dep.group == null || dep.version == null || dep.name == null || dep.name == "unspecified")
+                        return // ignore invalid dependencies
+
+                    final dependencyNode = dependenciesNode.appendNode('dependency')
+                    dependencyNode.appendNode('groupId', dep.group)
+                    dependencyNode.appendNode('artifactId', dep.name)
+                    dependencyNode.appendNode('version', dep.version)
+                    dependencyNode.appendNode('scope', scope)
+
+                    if (!dep.transitive) {
+                        // If this dependency is transitive, we should force exclude all its dependencies them from the POM
+                        final exclusionNode = dependencyNode.appendNode('exclusions').appendNode('exclusion')
+                        exclusionNode.appendNode('groupId', '*')
+                        exclusionNode.appendNode('artifactId', '*')
+                    } else if (!dep.properties.excludeRules.empty) {
+                        // Otherwise add specified exclude rules
+                        final exclusionNode = dependencyNode.appendNode('exclusions').appendNode('exclusion')
+                        dep.properties.excludeRules.each { ExcludeRule rule ->
+                            exclusionNode.appendNode('groupId', rule.group ?: '*')
+                            exclusionNode.appendNode('artifactId', rule.module ?: '*')
+                        }
+                    }
+                }
+
+                // List all "compile" dependencies (for old Gradle)
```

Comment:
do we need to support older versions of gradle? I'd drop this

Reply:
51bbdd0